### ISSUE TITLE
Allow collections to include other collections, recursively.

### ISSUE THREAD

**Issue by bredelings** [OpenTreeOfLife/opentree#930]
Now that we have so many collections, it would be useful to allow collections to contain other collections.

This would allow making e.g. a mammals collection, that contained a primates collection, and a bats collection, etc.

**Comment by jimallman** [comment]
I'm pondering this now, including some complications in the collections UI that will result. Here's some example JSON showing a collection in another one's `decisions` array:

```json
            {   //tree
                "name": "Tree tree2 from Dewey, 2015", 
                "comments": "from Dewey, 2015", 
                "decision": "INCLUDED", 
                "studyID": "tt_32", 
                "treeID": "tree2",
                "SHA": ""
            }, 
            {   //collection
                "name": "crabs", 
                "comments": "Brachyuran phylogenies", 
                "decision": "INCLUDED", 
                "collectionID": "kacrandall/crabs",
                "SHA": "a76e140b29db99c62bbb12bbb07c7778279dacef"
            }, 
```
Note that we can discern a member's type by the IDs included. If we decide to add a more explicit "type" property, we'll need to sweep the existing collections and update phylesystem.

**Comment by kcranston** [comment]
Would it be safer to have an explicit type, rather than relying on parsing the format of the ID?

I also see a SHA for a collection but not for trees. Is that designed inconsistency? Would we not default to the latest version of the collection?

**Comment by jimallman** [comment]
> Would it be safer to have an explicit type, rather than relying on parsing the format of the ID?

It does seem cleaner, but (as mentioned above) it means we'll need to sweep the existing collections and add these values programmatically to all the existing trees. Since the type _and number_ of IDs is different for trees vs. collections, I thought these properties made a reasonable marker for type. 

> I also see a SHA for a collection but not for trees. Is that designed inconsistency? Would we not default to the latest version of the collection?

I thought the default ("un-pinned") behavior is to follow the latest version on branch `master`. In this case, I'd leave this field empty. Do you think we should initially pin to the tree (or collection) version **at the time it's added**? I'm not sure which is likely to be more confusing in practice:

  - "Why aren't my latest fixes to this tree showing up in synthesis?"
  - "Why is synthesis different this time? (Oh, someone's been messing with my trees.)"

**Comment by kcranston** [comment]
Yes, default = latest SHA. I just wanted to make sure we weren't automatically including a specific SHA for a collection, since the example include a SHA for a collection but not a tree.

**Comment by jimallman** [comment]
Gotcha. This is on my mind, since I'm mocking up a simple UI for pinning to versions. Here's the read-only view, with a "version" indicator pointing to a specific version:

![screen shot 2016-04-18 at 1 36 22 pm](https://cloud.githubusercontent.com/assets/446375/14613532/cfadfd46-056a-11e6-896a-60c24dfb0b6c.png)

... and the default (un-pinned) state:

![screen shot 2016-04-18 at 1 36 02 pm](https://cloud.githubusercontent.com/assets/446375/14613546/dff34ee0-056a-11e6-9284-a3a81c72622b.png)

In either case, clicking on the icon would open a GitHub page for the chosen SHA (or latest master) in a new window. In the active editor, they'll be able to paste in a specific SHA, or "snapshot" the latest version, or clear this to use the default behavior.

**Comment by kcranston** [comment]
Not sure incorporating versions in the UI is a priority right now.

**Comment by jimallman** [comment]
OK. I thought perhaps with pending nested collections, reproducibility was becoming more important.

**Comment by kcranston** [comment]
The synthesis pipeline captures the SHAs used in synthesis, and we can do the same for any other analyses or downloads. So we have reproducibility covered. I am worried about complicating the UI.

**Comment by jimallman** [comment]
> Not sure incorporating versions in the UI is a priority right now.

OK, I'll leave my (very preliminary) work parked on this branch for now:
https://github.com/OpenTreeOfLife/opentree/compare/ui-for-pinning-collection-members
